Robot2018/src/org/usfirst/frc/team199/Robot2018/RobotMap.java

```diff
@@ -95,69 +93,60 @@ private void configSPX(WPI_VictorSPX mc) {
 	}
 
 	public RobotMap() {
-		
+
 		intakeMotor = new WPI_TalonSRX(getPort("IntakeTalonSRX", 4));
 		configSRX(intakeMotor);
 		liftMotor = new WPI_TalonSRX(getPort("LiftTalonSRX", 5));
 		configSRX(liftMotor);
 		climberMotor = new WPI_TalonSRX(getPort("ClimberTalonSRX", 6));
 		configSRX(climberMotor);
 
-		leftEnc = new Encoder(getPort("1LeftEnc", 0), getPort("2LeftEnc", 1));
+		leftEncPort1 = new DigitalInput(getPort("1LeftEnc", 0));
+		leftEncPort2 = new DigitalInput(getPort("2LeftEnc", 1));
+		leftEncDist = new Encoder(leftEncPort1, leftEncPort2);
+		leftEncDist.setPIDSourceType(PIDSourceType.kDisplacement);
+		leftEncRate = new Encoder(leftEncPort1, leftEncPort2);
+		leftEncRate.setPIDSourceType(PIDSourceType.kRate);
 		dtLeftDrive = new WPI_TalonSRX(getPort("LeftTalonSRXDrive", 0));
```

Comment:
I recommend changing the names of dtLeftDrive and dtRightDrive (and the associated pref names) to use the term "master" instead of "drive" to avoid confusion with the dtLeft and dtRight. Also, if there isn't a good reason to make the master and slave motor controller variables public, they should be private.